### PR TITLE
Add pytorch-3dunet

### PR DESCRIPTION
**Pull request by wolny**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by wolny** [comment]
@conda-forge/help-python this one is ready for review. Cheers!

File: recipes/pytorch-3dunet/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    - tensorboard
+    - tqdm
+    - h5py
+    - scipy
+    - scikit-image
+    - pyyaml
+  run_constrained:
+    - pytorch >=1.7.0
+
+test:
```
Comment (xhochy) [inline recipes/pytorch-3dunet/meta.yaml:35]:
```suggestion
test:
  requires:
    - pip
  commands:
    - pip check
```

File: recipes/pytorch-3dunet/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  sha256: 20bff5c53404d6b34b02a6a051cad80d93bc57646a76117e917951d34d2780dd
+
+build:
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+  number: 0
+  entry_points:
+    - predict3dunet = pytorch3dunet.predict:main
+    - train3dunet = pytorch3dunet.train:main
+
```
Comment (xhochy) [inline recipes/pytorch-3dunet/meta.yaml:18]:
```suggestion
```

File: recipes/pytorch-3dunet/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    - pip
+  run:
+    - python >=3.9
+    - tensorboard
+    - tqdm
+    - h5py
+    - scipy
+    - scikit-image
+    - pyyaml
+  run_constrained:
```
Comment (xhochy) [inline recipes/pytorch-3dunet/meta.yaml:32]: The package name sounds like it has a hard dependency on `pytorch`
```suggestion
```

**Review by xhochy — CHANGES_REQUESTED** [review]

File: recipes/pytorch-3dunet/meta.yaml (hunk 1 above)
Comment (wolny) [reply recipes/pytorch-3dunet/meta.yaml:35]: Previously, I've had `pytest tests` execution in here, but removed it due to build failing on win 64 (see my comment below).

Comment (xhochy) [reply recipes/pytorch-3dunet/meta.yaml:35]: Please add the pip check, we don't need pytest but `pip check` is essential for Python packages.

File: recipes/pytorch-3dunet/meta.yaml (hunk 3 above)
Comment (xhochy) [reply recipes/pytorch-3dunet/meta.yaml:32]: Build failures due to missing dependencies with noarch packages on windows is fine. We have that with all pytorch packages. Please bring in the hard dependency again and add the pip check.

**Comment by wolny** [comment]
Thanks again for the review @xhochy. I've addressed your comments and the recipe should be good to go :rocket:

**Review by xhochy — APPROVED** [review]

**Pull request merged**